src/Utility/Hash.php

```diff
@@ -1155,10 +1155,11 @@ public static function mergeDiff(array $data, array $compare): array
      *
      * @param array $data List to normalize
      * @param bool $assoc If true, $data will be converted to an associative array.
+     * @param mixed $default The default value to use when a top level numeric key is converted to associative form.
      * @return array
      * @link https://book.cakephp.org/4/en/core-libraries/hash.html#Cake\Utility\Hash::normalize
      */
-    public static function normalize(array $data, bool $assoc = true): array
+    public static function normalize(array $data, bool $assoc = true, $default = null): array
```

Comment:
I wonder if `default` is the right word since it's technically always the value not the fallback value.

Reply:
That's the var name we have used to specify the value to be used in place of a non-existent one, for e.g. InstanceConfigTrait::getConfig().